plugins/woocommerce/includes/admin/class-wc-admin-assets.php

```diff
@@ -554,6 +556,56 @@ public function admin_scripts() {
 			}
 		}
 
+		/**
+		 * Enqueue block editor assets.
+		 *
+		 * @return void
+		 */
+		public function enqueue_block_editor_assets() {
+			$settings_tabs = apply_filters('woocommerce_settings_tabs_array', []);
+
+			if ( is_array( $settings_tabs ) && count( $settings_tabs ) > 0  ) {
+				$formatted_settings_tabs = array();
+				foreach ($settings_tabs as $key => $label) {
+					$formatted_settings_tabs[] = array(
+						'key'   => is_string( $key ) ? $key : '',
+						'label' => is_string( $label ) ? $label : '',
+					);
+				}
+
+				WCAdminAssets::register_script( 'wp-admin-scripts', 'command-palette' );
+				wp_localize_script(
+					'wc-admin-command-palette',
+					'wcCommandPaletteSettings',
+					array(
+						'settingsTabs'    => $formatted_settings_tabs,
+					)
+				);
+			}
+
+			$admin_features_disabled = apply_filters( 'woocommerce_admin_disabled', false );
+			if ( ! $admin_features_disabled ) {
+				$analytics_reports = Analytics::get_report_pages();
+				if ( is_array( $analytics_reports ) && count( $analytics_reports ) > 0 ) {
+					$formatted_analytics_reports = array_map( function( $report ) {
+						return array(
+							'title' => is_string( $report['title'] ) ? $report['title']: '' ,
+							'path' => is_string( $report['path'] ) ? $report['path']: '',
+						);
+					}, $analytics_reports );
```

Comment:
@Aljullu - maybe use `array_filter()` here as some indexes have empty values which causes PHP warnings.

```
[29-Nov-2023 01:41:52 UTC] PHP Warning:  Trying to access array offset on value of type null in /Users/royho/projects/woocommerce-monorepo/plugins/woocommerce/includes/admin/class-wc-admin-assets.php on line 592
[29-Nov-2023 01:41:52 UTC] PHP Warning:  Trying to access array offset on value of type null in /Users/royho/projects/woocommerce-monorepo/plugins/woocommerce/includes/admin/class-wc-admin-assets.php on line 593
```

Reply:
Good idea! I added some extra protections in https://github.com/woocommerce/woocommerce/pull/41773, but wondering if you have some specific reproduction steps, so I can check whether that fixes the issue? In any case, I added you as the reviewer of that PR. Hope that's ok!